### PR TITLE
[staking] remove unnecessary call from GetStakingStateReader

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
as title

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/protocol/staking/candidate_statereader.go [hunk 1]
```diff
@@ -123,18 +123,6 @@ func (c *candSR) ActiveBucketsCount() uint64 {
 func GetStakingStateReader(sr protocol.StateReader) (CandidateStateReader, error) {
```
Comment (Liuhaai) [inline action/protocol/staking/candidate_statereader.go:123]: Replace `GetStakingStateReader` with `ConstructBaseView`?

File: action/protocol/staking/candidate_statereader.go [hunk 2]
```diff
@@ -123,18 +123,6 @@ func (c *candSR) ActiveBucketsCount() uint64 {
...
-			view, height, err := CreateBaseView(sr, true)
-			if err != nil {
-				return nil, err
-			}
-			return &candSR{
-				StateReader: sr,
-				height:      height,
-				view:        view,
-			}, nil
-		}
```
[the remark this answers is not in the document]
Comment (CoderZhi) [reply action/protocol/staking/candidate_statereader.go:137]: in this case, do we still need to keep this function?

[the remark this answers is not in the document]
Comment (dustinxie) [reply action/protocol/staking/candidate_statereader.go:137]: fixed

File: action/protocol/staking/candidate_statereader.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/staking/candidate_statereader.go]: fixed

**Review by Liuhaai — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

**Pull request merged**